kong/db/schema/init.lua

```diff
@@ -1777,6 +1778,52 @@ function Schema:process_auto_fields(data, context, nulls, opts)
               end
             end
 
+            if prev_refs and prev_refs[key] then
+              if refs then
+                if not refs[key] then
+                  refs[key] = prev_refs[key]
+                end
+
+              else
+                refs = { [key] = prev_refs[key] }
+              end
+            end
+          end
+
+        elseif vtype == "table" and ftype == "map" then
```

Comment:
could you add a test for this change, either one in spec/02-integration/03-db/11-db_transformations_spec.lua or a unit test?

Reply:
Added, thanks!